src/utils/TraceUtils.js

```diff
@@ -177,3 +170,73 @@ export const treeDecorate = (key, f, tree) => {
     newTree[key] = f(tree);
     return newTree;
 }
+
+export const highlightProperty = 'isHighlighted';
+
+// Takes a tree and attaches the property 'isHighlighted' to every node in the
+// tree. All those nodes whose trace has the given trace as prefix will get
+// 'isHiglighted' = true, and the rest false.
+export const forwardMatching = (trace, tree) => {
+    let f;
+    if (trace === null) {
+        // Completely null traces do not match anything.
+        f = (t) => (false);
+    } else {
+        // If the given trace is a subtrace of the trace of a node, it should get
+        // highlighted.
+        f = (t) => (containsSubtrace(trace, t.tra))
```

Comment:
When structuredLang is merged, the property will be called `trace` instead of `tra`.

Also:
* Nit: remove parentheses around `containsSubtrace(...)` -- they are not needed.
* Nit: add semicolon to end of line.